### ISSUE TITLE
root class is not shown as autocompletion option for generics

### ISSUE THREAD

**Issue by EIA485** [EmmyLuaLs/emmylua-analyzer-rust#685]
please tell me if I missed some configuration option.
When using generics, I seem to only get derived classes as suggestions. manually typing BaseClass as the argument does not show a warning.
```Lua 
---@class BaseClass

---@class DerivedClass : BaseClass

---@generic T : BaseClass
---@param Arg1 `T`
---@return T
local function create(Arg1)
  Return spawn(Arg1)
End


--Typing
create("") -- only suggests "DerivedClass"
```
